micros/modules/networking/firewall.nix

```diff
@@ -0,0 +1,430 @@
+          See the description at networking.firewall.connectionTrackingModules
+
+          (needs kernel 3.5+)
+        '';
+      };
+      extraInputRules = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        example = "ip6 saddr { fc00::/7, fe80::/10 } tcp dport 24800 accept";
+        description = ''
+          Additional nftables rules to be appended to the input-allow
+          chain.
+
+          This option only works with the nftables based firewall.
+        '';
+      };
+
+      extraForwardRules = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        example = "iifname wg0 accept";
+        description = ''
+          Additional nftables rules to be appended to the forward-allow
+          chain.
+
+          This option only works with the nftables based firewall.
+        '';
+      };
+
+      extraReversePathFilterRules = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        example = "fib daddr . mark . iif type local accept";
+        description = ''
+          Additional nftables rules to be appended to the rpfilter-allow
+          chain.
+
+          This option only works with the nftables based firewall.
+        '';
+      };
+      interfaces = lib.mkOption {
+        default = {};
+        type = with lib.types; attrsOf (submodule [{options = interfaceOpts;}]);
+        description = ''
+          Interface-specific open ports.
+        '';
+      };
+
+      allInterfaces = lib.mkOption {
+        internal = true;
+        visible = false;
+        default =
+          {
+            default = lib.mapAttrs (name: value: cfg.${name}) interfaceOpts;
+          }
+          // cfg.interfaces;
+        type = with lib.types; attrsOf (submodule [{options = interfaceOpts;}]);
+        description = ''
+          All open ports.
+        '';
+      };
+    }
+    // interfaceOpts;
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.extraCommands == "";
+        message = "extraCommands is incompatible with the nftables based firewall: ${cfg.extraCommands}";
+      }
+      {
+        assertion = cfg.extraStopCommands == "";
+        message = "extraStopCommands is incompatible with the nftables based firewall: ${cfg.extraStopCommands}";
+      }
+      {
+        assertion = cfg.pingLimit == null || !(lib.hasPrefix "--" cfg.pingLimit);
+        message = "nftables syntax like \"2/second\" should be used in networking.firewall.pingLimit";
+      }
+      {
+        assertion = config.networking.nftables.rulesetFile == null;
+        message = "networking.nftables.rulesetFile conflicts with the firewall";
+      }
+    ];
+
+    networking.nftables.tables."nixos-fw".family = "inet";
+    networking.nftables.tables."nixos-fw".content = ''
+      set temp-ports {
+        comment "Temporarily opened ports"
+        type inet_proto . inet_service
+        flags interval
+        auto-merge
+      }
```

Comment:
Nixpkgs adds this alongside a helper utility to add or remove ports. If we're inheriting this, we should either get or rewrite the firewall utility.